node_packages/bazed-svelte/src/lib/Rekuho.svelte

```diff
@@ -77,7 +80,32 @@
   $: lineCount = Math.ceil(height / lineHeight)
   $: columnCount = Math.ceil(width / columnWidth)
 
-  const transformToScreenPosition = ([x, y]: Vector2): Vector2 => [x * columnWidth, y * lineHeight]
+  const transformToScreenPosition = ([x, y]: Vector2): ScreenPosition => {
+    return {
+      col: x * columnWidth,
+      line: y * lineHeight,
+    }
+  }
+
+  const transformToSelection = (
+    caret: CoordinateRegion,
+  ): { start: ScreenPosition; end: ScreenPosition } => {
+    let start = caret.head
+    let end = caret.tail
+
+    if (
+      caret.head.line > caret.tail.line ||
+      (caret.head.line === caret.tail.line && caret.head.col > caret.tail.col)
+    ) {
+      start = caret.tail
+      end = caret.head
+    }
+
+    return {
+      start,
+      end,
+    }
+  }
```

Comment:
I feel like we at some point want to have some better separation  for this sort of logic to stay somewhere else, and to have positional comparison be somewhere else, too, but for now this is fine!

Reply:
yep, agreed

Reply:
So true!

Reply:
don't reopen the conversation to shitpost

Reply:
Hey, I was playing around with plugins, not shitposting. :(